ref-exchange/src/account_deposit.rs

```diff
@@ -5,29 +5,100 @@ use std::convert::TryInto;
 
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::json_types::{ValidAccountId, U128};
-use near_sdk::{assert_one_yocto, env, near_bindgen, AccountId, Balance, PromiseResult};
+use near_sdk::{
+    assert_one_yocto, env, near_bindgen, AccountId, Balance, PromiseResult, StorageUsage,
+};
 
 use crate::utils::{ext_fungible_token, ext_self, GAS_FOR_FT_TRANSFER};
 use crate::*;
 
-const MAX_ACCOUNT_LENGTH: u128 = 64;
-const MIN_ACCOUNT_DEPOSIT_LENGTH: u128 = MAX_ACCOUNT_LENGTH + 16 + 4;
+const U128_STORAGE: StorageUsage = 16;
+/// bytes length of u64 values
+const U64_STORAGE: StorageUsage = 8;
+/// bytes length of u32 values. Used in length operations
+const U32_STORAGE: StorageUsage = 4;
+/// max length of account id * 8 (1 byte)
+const ACC_ID_STORAGE: StorageUsage = 64 * 8;
+// 64 = max account name length
+
+// ACC_ID: the Contract accounts map
+// + U128_STORAGE: near_amount storage
+// + U32_STORAGE: tokens HashMap length
+// + U64_STORAGE: storage_used
+// + 2: version
+pub const INIT_ACCOUNT_STORAGE: StorageUsage =
+    ACC_ID_STORAGE + U128_STORAGE + U32_STORAGE + 2 * U64_STORAGE + 2;
+
+// NEAR native token. This is not a valid token ID. HACK: NEAR is a native token, we use the
+// empty string we use it to reference not existing near account.
+// pub const NEAR: AccountId = "".to_string();
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub enum AccountDeposit {
+    V2(AccountDepositV2),
+}
+
+impl From<AccountDeposit> for AccountDepositV2 {
+    fn from(account: AccountDeposit) -> Self {
+        match account {
+            AccountDeposit::V2(a) => {
+                if a.storage_used > 0 {
+                    return a;
+                }
+                // migrate from V1
+                a.storage_used = U64_STORAGE;
+                a
+            }
+        }
+    }
+}
 
 /// Account deposits information and storage cost.
+/// Legacy version
 #[derive(BorshSerialize, BorshDeserialize, Default)]
 #[cfg_attr(feature = "test", derive(Clone))]
-pub struct AccountDeposit {
-    /// Native amount sent to the exchange.
-    /// Used for storage now, but in future can be used for trading as well.
-    pub amount: Balance,
+pub struct AccountDepositV1 {
+    /// NEAR sent to the exchange.
+    /// Used for storage and trading.
+    pub near_amount: Balance,
     /// Amounts of various tokens in this account.
     pub tokens: HashMap<AccountId, Balance>,
 }
 
-impl AccountDeposit {
+/// Account deposits information and storage cost.
+#[derive(BorshSerialize, BorshDeserialize, Default)]
+#[cfg_attr(feature = "test", derive(Clone))]
+pub struct AccountDepositV2 {
```

Comment:
let's rename to Account?

Reply:
The v1 structure is already named AccountDeposit. So - do you suggest to rename both?